src/psyclone/domain/lfric/kernel_arg_order.py

```diff
@@ -0,0 +1,1063 @@
+                            3         W1, W2, W2H, W2V, W2broken, ANY_W2
+
+        Differential Basis  1         W2, W2H, W2V, W2broken, ANY_W2
+
+                            3         W0, W1, W2trace, W2Htrace, W2Vtrace,
+                                      W3, Wtheta, Wchi
+
+        :param str name: 'basis' or 'diff_basis'.
+        :param str function_space: the current function space.
+
+        raises InternalError: if unexpected shape metadata is found.
+
+        '''
+        function_space_name = self._function_space_name(function_space)
+        const = LFRicConstants()
+        if not self._metadata.shapes:
+            return
+        for shape in self._metadata.shapes:
+            if shape in const.VALID_QUADRATURE_SHAPES:
+                self.arg_info.append(
+                    f"{name}_{function_space_name}_qr_{shape.split('_')[-1]}")
+                self._arg_index += 1
+            elif shape in const.VALID_EVALUATOR_SHAPES:
+                if self._metadata.evaluator_targets:
+                    target_function_spaces = self._metadata.evaluator_targets
+                else:
+                    # Targets are the function spaces of all modified fields.
+                    fields = [field for field in self._metadata.meta_args
+                              if type(field) in [
+                                      FieldArgMetadata, FieldVectorArgMetadata,
+                                      InterGridArgMetadata,
+                                      InterGridVectorArgMetadata]
+                              and field.access != "gh_read"]
+                    target_function_spaces = []
+                    for field in fields:
+                        if field.function_space not in target_function_spaces:
+                            target_function_spaces.append(field.function_space)
+                for target_function_space in target_function_spaces:
+                    self.arg_info.append(
+                        f"{name}_{function_space_name}_to_"
+                        f"{target_function_space}")
+                    self._arg_index += 1
+            else:
+                raise InternalError(
+                    f"Unexpected shape metadata. Found '{shape}' but expected "
+                    f"one of {const.VALID_EVALUATOR_SHAPES}.")
+
+    def basis(self, function_space):
+        '''Arguments associated with basis functions on the supplied function
+        space.
+
+        :param str function_space: the current function space.
+
+        '''
+        self._basis_or_diff_basis("basis", function_space)
+
+    def diff_basis(self, function_space):
+        '''Arguments associated with differential basis functions on the
+        supplied function space.
+
+        :param str function_space: the current function space.
+
+        '''
+        self._basis_or_diff_basis("diff_basis", function_space)
+
+    def quad_rule(self, shapes):
+        '''Quadrature information is required (gh_shape =
```

Comment:
Again, can we use this in the UG do you think?

Reply:
Yes possibly, However, these descriptions are no longer in this PR.